cli/src/test/all-tests.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+#set -eu
+set -o pipefail
+
+source "$(dirname "${0}")"/functions-test.sh
+
+MATRIX_OS="$1"
+# Switch IDEasy binary file name based on github workflow matrix.os name (first argument of all-tests.sh)
+BINARY_FILE_NAME="ideasy"
+if [ "${MATRIX_OS}" == "windows-latest" ]; then
+  BINARY_FILE_NAME="ideasy.exe"
+fi
+
+START_TIME=$(date '+%Y-%m-%d_%H-%M-%S')
+
+DEBUG_INTEGRATION_TEST_PREFIX="${HOME}/tmp/ideasy-integration-test-debug"
+DEBUG_INTEGRATION_TEST="${DEBUG_INTEGRATION_TEST_PREFIX}-${START_TIME}"
+IDE_HOME="${DEBUG_INTEGRATION_TEST}/home-dir"
+export IDE_ROOT="${IDE_HOME}/projects"
+IDEASY_DIR="${IDE_ROOT}/_ide"
+FUNCTIONS="${IDEASY_DIR}/installation/functions"
+IDE="${DEBUG_INTEGRATION_TEST}/home-dir/projects/_ide/bin/${BINARY_FILE_NAME}"
+TEST_RESULTS_FILE="${IDE_ROOT}/testResults"
+
+test_files_directory=$(realpath "$0" | xargs dirname)
+
+success=0
+failure=0
+total=0
+
+function doTestsInner() {
+  # Note: requires var test_files_directory to be set.
+  echo "Entering doTestsInner..."
+  for testpath in "${test_files_directory:?}/integration-tests"/*; do
+    testcase="${testpath/*\//}"
+    echo "Running test #${total}: ${testcase} (${testpath})"
+
+    integration_test_result=0
+
+    # Following line adds a trap to ERR signal: Whenever error (some
+    # conditions apply; see documentation) set integration_test_result=1.
+    trap 'doWarning "A non-handled error in integration test occurred."; integration_test_result=1' ERR
+
+    source "${testpath:?}"
```

Comment:
if one test "crashes" then the entire test-suite will "crash" and next tests will not be executed?
Or is that handled by the `trap`?
If so - great solution but then put stuff like `set -e` here instead of copy&pasting it into each individual test.
Otherwise we should include/source a common "script" in all tests so we can later extend things without refactoring every individual test.

Reply:
I've moved all `set -e` from the individual tests to the function scripts now instead.

Reply:
So your answer to my first question is "yes"? Then I am happy 😄